src/environmentsPage.js

```diff
@@ -0,0 +1,260 @@
+/*
+ * This file is part of PROS Application Browser, a flash enabled
+ * browser restricted to preconfigured servers.
+ * Copyright (C) 2018 PROS, Inc.
+ *
+ * PROS Application Browser is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PROS Application Browser is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PROS Application Browser.  If not, see http://www.gnu.org/licenses/
+ * You can contact PROS, Inc. with any questions at http://www.pros.com
+ */
+const fs = require('fs');
+
+/**
+ * Set the path to the config file based on platform
+ */
+const CONFIG_FILE_PATH =
+  process.platform === 'darwin'
+    ? '/pros/desktopclient/config.json'
+    : 'config.json';
+
+const SETTINGS_FILE_PATH =
+  process.platform === 'darwin'
+    ? '/pros/desktopclient/launchpage-config.json'
+    : process.env.USERPROFILE + '\\launchpage-config.json';
+
+const getSyntaxError = (error) => {
+  const isExplicit = error instanceof SyntaxError;
+  return `[${isExplicit ? 'EXPLICIT' : 'IMPLICIT'}] ${error.name} in config.json: ${error.message}`;
+};
+
+const formatError = message => `<br/><li>${message}</li>`;
```

Comment:
I reviewed the use cases and don't see a direct issue here, but we need to sanitize message for safety